### ISSUE TITLE
Merge tests/drivers/adc/adc_api and adc_dma in a single test

### ISSUE THREAD

**Issue by gautierg-st** [zephyrproject-rtos/zephyr#68690]
**Is your enhancement proposal related to a problem? Please describe.**

I'd like to expand test support for ADC DMA feature for STM32 following this PR: #68685
But the tests/drivers/adc/adc_dma test still configures the ADC channels in the source code instead of in dts overlays (contrary to adc_api that was updated in #56647).
By looking at both tests (and apart from the dts/source code channel config), they seem pretty identical in what they do and test, the difference is that adc_dma enables the DMA (obviously!).

**Describe the solution you'd like**

Since both tests are practically the same, I think it would be a good idea to add a new .conf file and a new test case inside adc_api testcase.yaml that would test the ADC with DMA.
That way, we could remove the duplicate test adc_dma, and reuse the many dts overlays already defined in adc_api by just adding the DMA configuration to them.

**Describe alternatives you've considered**

Another solution could be to port adc_dma test to use the dts overlays, just like adc_api. But the test itself would practically be a duplicate of adc_api, and we would also have to duplicate all overlays.

@anangl I'd be happy to do the change, but I'd like to hear from you before starting on it. Do you think this proposal is acceptable?

**Comment by gautierg-st** [comment]
Great, I'll work on it after v3.6.0 is out and the HW model v2 is merged!
